subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/MqttNotificationServiceTest.kt

```diff
@@ -0,0 +1,138 @@
+        entities = emptySet(),
+        notification = NotificationParams(
+            attributes = emptyList(),
+            endpoint = Endpoint(
+                uri = "mqtt://test@localhost:1883/notification".toUri(),
+                notifierInfo = listOf(
+                    EndpointInfo(Mqtt.Version.KEY, Mqtt.Version.V5)
+                )
+            )
+        ),
+        contexts = emptyList()
+    )
+    private val validMqttNotificationData = MQTTNotificationData(
+        brokerUrl = "tcp://localhost:1883",
+        clientId = "clientId",
+        username = "test",
+        topic = "/notification",
+        qos = 0,
+        mqttMessage = MQTTNotificationData.MqttMessage(getNotificationForSubscription(mqttSubscription), emptyMap())
+    )
+
+    private fun getNotificationForSubscription(subscription: Subscription) = Notification(
+        subscriptionId = subscription.id,
+        data = listOf(mapOf("hello" to "world"))
+    )
+
+    @Test
+    fun `mqttNotifier should process endpoint uri to get connexion information`() = runTest {
+        val subscription = mqttSubscription
+        coEvery { mqttVersionService.callMqttV3(any()) } returns Unit
+        assert(
+            mqttNotificationService.mqttNotifier(
+                subscription,
+                getNotificationForSubscription(subscription),
+                mapOf()
+            )
+        )
+
+        coVerify {
+            mqttVersionService.callMqttV3(
+                match {
+                    it.username == validMqttNotificationData.username &&
+                        it.password == validMqttNotificationData.password &&
+                        it.topic == validMqttNotificationData.topic &&
+                        it.brokerUrl == validMqttNotificationData.brokerUrl
+                }
+            )
+        }
+    }
+
+    @Test
+    fun `mqttNotifier should use notifier info to choose the mqtt version`() = runTest {
+        val subscription = mqttSubscription
+        coEvery { mqttVersionService.callMqttV3(any()) } returns Unit
+        coEvery { mqttVersionService.callMqttV5(any()) } returns Unit
+
+        mqttNotificationService.mqttNotifier(
+            subscription,
+            getNotificationForSubscription(subscription),
+            mapOf()
+        )
+        coVerify(exactly = 1) {
+            mqttVersionService.callMqttV3(
+                any()
+            )
+        }
+        coVerify(exactly = 0) {
+            mqttVersionService.callMqttV5(
+                any()
+            )
+        }
+
+        mqttNotificationService.mqttNotifier(
+            mqttSubscriptionV5,
+            getNotificationForSubscription(subscription),
+            mapOf()
+        )
+        coVerify(exactly = 1) {
+            mqttVersionService.callMqttV5(
+                any()
+            )
+        }
+        coVerify(exactly = 1) {
+            mqttVersionService.callMqttV3(
+                any()
+            )
```

Comment:
is this to verify the first call using V3 that was made in the begining ?

Reply:
its to verify it was not called a second time